### ISSUE TITLE
expose Github Actions CI artefacts

### ISSUE THREAD

**Issue by Nelson-numerical-software** [PortAudio/portaudio#558]
It could be great (specially for Windows platforms) to export Github actions artefacts. (.dll, .lib, .h)

```
 - uses: actions/upload-artifact@v1
      with:
        name: github-action-artifacts
        path: artifacts/
```

**Comment by philburk** [comment]
This sounds interesting. What do you think would be the primary purpose? Debugging, testing releases, obtaining precompiled binaries?

I found these helpful links:
https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts
https://docs.github.com/en/actions/managing-workflow-runs/downloading-workflow-artifacts

**Comment by Nelson-numerical-software** [comment]
testing prereleases and precompiled binaries ;)

**Comment by philburk** [comment]
We have some other big CI changes merging soon. Let's do this after they merge.

**Comment by RossBencina** [comment]
This sounds like "release binaries for windows" by another name. I don't think we should do it. We don't have the resources to support binary builds -- there are too many things that can go wrong.

**Comment by Nelson-numerical-software** [comment]
> This sounds like "release binaries for windows" by another name. I don't think we should do it. We don't have the resources to support binary builds -- there are too many things that can go wrong.

It is better for the official repository to do this than someone else to do the same in a fork.

Portaudio is a very good library but on Windows, portaudio does not help advanced users on Windows with visual studio.

It is not "easy" to nuild (dependencies) and the most important for users is not to compile portaudio but to try to use it.

Currently, it seems that portaudio  does not provide binaries for visual studio and it is a real bottleneck on windows.
If portaudio does not want to support visual studio, please say it officially on your repository, users will search others alternatives.

Thanks for your work on PortAudio (used since more 10 years in some projects)

**Comment by dechamps** [comment]
> It is not "easy" to nuild (dependencies)

Really? Unless you want ASIO support, building PortAudio on Windows is a simple as opening the folder in Visual Studio 2019 and hitting "build" (using the built-in CMake support). That's it. Literally. If you can't do that, you won't be able to build an application that uses PortAudio either, anyway.

**Comment by Nelson-numerical-software** [comment]
Happy to see that finally portaudio 19.7 is available in vcpkg.

You could add at least in each release a binary version for windows.

An example here from another great project:
https://github.com/libsndfile/libsndfile/releases

**Comment by Be-ing** [comment]
This is implemented in #461. I made GitHub Actions artifacts for the directory that CMake installs to.

**Comment by Be-ing** [comment]
This can be closed now that #461 was merged.

**Comment by RossBencina** [comment]
We will not release build artefacts. Closing.
